### PR TITLE
Hunain/issue 13(Partial),14/barcode Scanner

### PR DESCRIPTION
**Pull request by HunnySajid**
Related Task: 
[Issue-14](https://github.com/roots-id/roots-react-native/issues/14)
[Issue-13](https://github.com/roots-id/roots-react-native/issues/13) partially resolved with [1 issue](https://github.com/roots-id/roots-react-native/issues/13#issuecomment-1291256853), will look into it

Changelog:
- Added bar code scanning packages
- Dummy functionality to add new credential when a barcode is scanned

### CODE REVIEW

File: src/screens/ScanQrCodeScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    );
+  }
+
+  const handleDemo = async () => {
+    if (!scanned && configService.getDemo()) {
+      setScanned(true);
+      console.log('Scan QR - pretending to scan with demo data');
+      alert('No data scanned, using demo data instead.');
+      
+      if (modelType === 'contact') {
```
Comment (2byrds) [inline src/screens/ScanQrCodeScreen.tsx:61]: When you hit the scan button from the contact list, the dummy data should be a contact that is added to the contact list.  Similar to https://github.com/roots-id/rootswallet/blob/main/src/screens/ScanQRCodeScreen.tsx#L36

**Review by 2byrds — CHANGES_REQUESTED** [review]

**Comment by 2byrds** [comment]
Looks pretty good so far!  is there a camera component that we can use when web is detected?

**Comment by 2byrds** [comment]
ios scan modal is expectedly dark on simulator.  i'll test it with a physical device soon
image

File: src/screens/ScanQrCodeScreen.tsx (hunk 1 above)
Comment (2byrds) [reply src/screens/ScanQrCodeScreen.tsx:61]: Also, I see errors when I scan real QR codes.  On scan can we post a console message saying scan complete, but only using mock data?
` WARN  Possible Unhandled Promise Rejection (id: 1):
SyntaxError: JSON Parse error: Unexpected identifier "https"
parse@[native code]
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:182911:34
generatorResume@[native code]
asyncGeneratorStep@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:25017:26
_next@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:25039:29
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:25046:14
tryCallTwo@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:30558:9
doResolve@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:30722:25
Promise@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:30581:14
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:25035:25
handleBarCodeScanned@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:182924:27
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:183086:21
invokeGuardedCallbackProd@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3109:21
invokeGuardedCallback@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3213:42
invokeGuardedCallbackAndCatchFirstError@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3217:36
executeDispatch@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3293:48
executeDispatchesInOrder@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3313:26
executeDispatchesAndRelease@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4662:35
forEachAccumulated@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:3845:18
runEventsInBatch@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4686:27
runExtractedPluginEventsInBatch@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4766:25
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4741:42
batchedUpdates$1@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:17151:20
batchedUpdates@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4648:36
_receiveRootNodeIDEvent@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4740:23
receiveEvent@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:4770:32
__callFunction@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:23976:36
http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:23704:31
__guard@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:23930:15
callFunctionReturnFlushedQueue@http://192.168.86.23:19000/node_modules/expo/AppEntry.bundle?platform=android&dev=true&hot=false:23703:21
callFunctionReturnFlushedQueue@[native code]
`

Comment (HunnySajid) [reply src/screens/ScanQrCodeScreen.tsx:61]: weird I did not notice that. Will take a look

**Comment by HunnySajid** [comment]
Hey @2byrds 

The following changes have been made, please verify:
- Add a new contact when barcode is scanned from contact list screen
- No crash on a QR code scan
- A console message is added to show that dummy credential/contact is scanned
- A camera component is added for web, now we dont see any compatibility message, a camera is opened on web instead. But I was not able to scan barcode/qr with that. 
Maybe it is because of the Mac camera? Not sure about that.

**Comment by 2byrds** [comment]
Confirmed scanning barcodes or not scanning anything until timeout generates mock data.  However the new contact has no welcome messages.  We can work on that later as part of the mock workfliw behavior

**Review by 2byrds — APPROVED** [review]

**Comment by 2byrds** [comment]
confirmed in ios and web as well

**Pull request merged**